sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/OperatorsShouldBeOverloadedConsistently.cs

```diff
@@ -0,0 +1,132 @@
+                {
+                    return;
+                }
+
+                var classMethods = classSymbol
+                    .GetMembers()
+                    .OfType<IMethodSymbol>()
+                    .Where(m => !m.IsConstructor())
+                    .ToList();
+
+                var implementedMethods = new HashSet<string>();
+                implementedMethods.Add(GetNameOrDefault(MethodName.OperatorPlus, classMethods, KnownMethods.IsOperatorBinaryPlus));
+                implementedMethods.Add(GetNameOrDefault(MethodName.OperatorMinus, classMethods, KnownMethods.IsOperatorBinaryMinus));
+                implementedMethods.Add(GetNameOrDefault(MethodName.OperatorEquals, classMethods, KnownMethods.IsOperatorEquals));
+                implementedMethods.Add(GetNameOrDefault(MethodName.OperatorNotEquals, classMethods, KnownMethods.IsOperatorNotEquals));
+                implementedMethods.Add(GetNameOrDefault(MethodName.ObjectEquals, classMethods, KnownMethods.IsObjectEquals));
+                implementedMethods.Add(GetNameOrDefault(MethodName.ObjectGetHashCode, classMethods, KnownMethods.IsObjectGetHashCode));
+                implementedMethods.Remove(null);
+
+                var requiredMethods = new HashSet<string>();
+
+                if (implementedMethods.Contains(MethodName.OperatorPlus))
+                {
+                    requiredMethods.Add(MethodName.OperatorMinus);
+                    requiredMethods.Add(MethodName.OperatorEquals);
+                    requiredMethods.Add(MethodName.OperatorNotEquals);
+                    requiredMethods.Add(MethodName.ObjectEquals);
+                    requiredMethods.Add(MethodName.ObjectGetHashCode);
+                }
+
+                if (implementedMethods.Contains(MethodName.OperatorMinus))
+                {
+                    requiredMethods.Add(MethodName.OperatorPlus);
+                    requiredMethods.Add(MethodName.OperatorEquals);
+                    requiredMethods.Add(MethodName.OperatorNotEquals);
+                    requiredMethods.Add(MethodName.ObjectEquals);
+                    requiredMethods.Add(MethodName.ObjectGetHashCode);
+                }
+
+                if (implementedMethods.Contains(MethodName.OperatorEquals))
+                {
+                    requiredMethods.Add(MethodName.OperatorNotEquals);
+                    requiredMethods.Add(MethodName.ObjectEquals);
+                    requiredMethods.Add(MethodName.ObjectGetHashCode);
+                }
+
+                if (implementedMethods.Contains(MethodName.OperatorNotEquals))
+                {
+                    requiredMethods.Add(MethodName.OperatorEquals);
+                    requiredMethods.Add(MethodName.ObjectEquals);
+                    requiredMethods.Add(MethodName.ObjectGetHashCode);
+                }
+
+                var missingMethods = requiredMethods.Except(implementedMethods);
+                if (missingMethods.Any())
```

Comment:
If you are using the `Except` of `HashSet` you probably have a `Count`/`Length` property instead of this call to `Any()`.